### PR TITLE
MTTL-3104: Add Processes classes

### PR DESCRIPTION
**Pull request by notatiyyah**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/MTTL-3104

## Describe this PR

Add Processes data classes so that we can build a processes search endpoint for MMH's worktray.

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

**Review by humulla — COMMENTED** [review]
Why are we adding the processes domain models here? Can we not use the process shared nuget package instead?

**Comment by notatiyyah** [comment]
> Why are we adding the processes domain models here? Can we not use the process shared nuget package instead?

@humulla  Forgot to add comments, this is a temporary measure till the process shared package is ready

File: Hackney.Shared.HousingSearch/Gateways/Models/Processes/QueryablePatchAssignment.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+using Nest;
+
+namespace Hackney.Shared.HousingSearch.Gateways.Models.Processes
+{
+    public class QueryablePatchAssignment
+    {
+        [Text(Name = "patchId")]
+        public string PatchId { get; set; }
```
Comment (humulla) [inline Hackney.Shared.HousingSearch/Gateways/Models/Processes/QueryablePatchAssignment.cs:8]: I think this should be a guid

File: Hackney.Shared.HousingSearch/Gateways/Models/Processes/QueryablePatchAssignment.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+        [Text(Name = "patchName")]
+        public string PatchName { get; set; }
+
+        [Text(Name = "responsibleEntityId")]
+        public string ResponsibleEntityId { get; set; }
```
Comment (humulla) [inline Hackney.Shared.HousingSearch/Gateways/Models/Processes/QueryablePatchAssignment.cs:14]: I think this should be a guid

File: Hackney.Shared.HousingSearch/Gateways/Models/Processes/QueryableProcess.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace Hackney.Shared.HousingSearch.Gateways.Models.Processes
+{
+    public class QueryableProcess
+    {
+
+        [Text(Name = "id")]
+        public string Id { get; set; }
```
Comment (humulla) [inline Hackney.Shared.HousingSearch/Gateways/Models/Processes/QueryableProcess.cs:10]: I think this should be a guid

File: Hackney.Shared.HousingSearch/Gateways/Models/Processes/QueryableRelatedEntity.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+using Hackney.Shared.HousingSearch.Domain.Process;
+using Nest;
+
+namespace Hackney.Shared.HousingSearch.Gateways.Models.Processes
+{
+    public class QueryableRelatedEntity
+    {
+        [Text(Name = "id")]
+        public string Id { get; set; }
```
Comment (humulla) [inline Hackney.Shared.HousingSearch/Gateways/Models/Processes/QueryableRelatedEntity.cs:9]: I think this should be a guid

File: Hackney.Shared.HousingSearch/Domain/Process/PatchAssignment.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+// TODO: Replace with class from Hackney.Shared.Processes
+
+namespace Hackney.Shared.HousingSearch.Domain.Process
+{
+    public class PatchAssignment
+    {
+        public string PatchId { get; set; }
```
Comment (humulla) [inline Hackney.Shared.HousingSearch/Domain/Process/PatchAssignment.cs:7]: I think this should be a guid

File: Hackney.Shared.HousingSearch/Domain/Process/PatchAssignment.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+        public string PatchName { get; set; }
+        public string ResponsibleEntityId { get; set; }
```
Comment (humulla) [inline Hackney.Shared.HousingSearch/Domain/Process/PatchAssignment.cs:9]: I think this should be a guid

File: Hackney.Shared.HousingSearch/Gateways/Models/Processes/QueryablePatchAssignment.cs (hunk 1 above)
Comment (notatiyyah) [reply Hackney.Shared.HousingSearch/Gateways/Models/Processes/QueryablePatchAssignment.cs:8]: In the other classes these are strings, not sure why, but I kept it like that to keep it consistent.

**Review by humulla — APPROVED** [review]

**Pull request merged**